02_ml_models/README.md

```diff
@@ -0,0 +1,77 @@
+* [ml5.js: Webcam Image Classification - video tutorial](https://youtu.be/D9BoBSkLvFo?list=PLRqwX-V7Uu6YPSwT06y_AEYTqIwbeam3y)
+* [ml5.js: Transfer Learning with Feature Extractor - video tutorial](https://youtu.be/kRpZ5OqUY6Y?list=PLRqwX-V7Uu6YPSwT06y_AEYTqIwbeam3y)
+* [ml5.js: Feature Extractor Classification - video tutorial](https://youtu.be/eeO-rWYFuG0?list=PLRqwX-V7Uu6YPSwT06y_AEYTqIwbeam3y)
+* [ml5.js: Feature Extractor Regression - video tutorial](https://youtu.be/aKgq0m1YjvQ?list=PLRqwX-V7Uu6YPSwT06y_AEYTqIwbeam3y)
+
+### ml5.js resources
+* [ml5 website](https://ml5js.org)
+* [ml5 web editor examples](https://github.com/ml5js/ml5-examples/)
+
+### ml5 code editor examples
+* [Drag and Drop Image Classification](https://editor.p5js.org/ml5/sketches/S1PLinE27)
+* [Webcam Image Classification](https://editor.p5js.org/ml5/sketches/IlF1JFvWjc)
+* [Webcam Image Classification Text to Speech](https://editor.p5js.org/ml5/sketches/Ry7EL4JvA3)
+* [Image Classification Multiple Images](https://editor.p5js.org/ml5/sketches/f3rqIqNey5)
+* [Image Classification Scavenger Hunt](https://editor.p5js.org/ml5/sketches/APzpeXOuEQ)
+
+### Defining Machine Learning yet again
+* "A computer program is said to learn from experience E with respect to some class of tasks T and performance measure P, if its performance at tasks in T, as measured by P, improves with experience E." -- Tom Mitchell (1998): [Machine Learning book](http://amzn.to/2nLdRgQ).
+  * Example: classifying images of dogs and cats.
+    * E = Watching you classify images as dogs or cats.
+    * T = Classifying images as dogs or cats.
+    * P = The % of images correctly classified.
+
+### Classification and Regression
+* Classification and regression both involve making a "prediction" based on input data.
+* Classification refers to predicting an output with a discrete set of possibilities like a set of categories or labels. For example: "Given an input image, is it a dog or cat?"
+* Regression refers to predicting an "continuous" output (a fancy way of saying number). For example: "Given the number of bedrooms, what is the price of a house?" or "Given an input image of a cat, how much does the cat weigh?"
+
+### Assignment:
+1. Watch [But what *is* a Neural Network?](https://youtu.be/aircAruvnKk) by 3Blue1Brown
+2. Read Andrey Kurenkov's ['Brief' History of Neural Nets and Deep Learning](http://www.andreykurenkov.com/writing/a-brief-history-of-neural-nets-and-deep-learning/)
+3. Read [ImageNet: The Data That Transformed AI Research—and Possibly the World](https://qz.com/1034972/the-data-that-changed-the-direction-of-ai-research-and-possibly-the-world/) by Dave Gershgorn (Note: Fei-Fei Li is no longer at Google; she is currently Co-Director of the Stanford Human-Centered AI Institute)
+4. Explore [ImageNet](http://image-net.org/index)
+5. TBA (exercise related to running ml5 image webcam example and/or teachable machine training -- no code writing just yet.)
```

Comment:
If the goal is for students to start coding after Session 2 this week, then what will best prepare students? Does it make sense / is there enough time on Monday to open up the p5 editor together, run one of the classification examples, go thru the code line by line (also thinking about Objectives #5 and #6 here), and use this in-class event to prompt the short (but no-coding) exercise(s) to complete before Wednesday?  

...and why not both exercises?! An exercise to learn more about MobileNet: for example, how many items does it correctly identify in your home / dorm? If it recognizes a banana, does the banana always have to be in the same position and with the same lighting? 

And another exercise to preview Session 2 and get creative with teachable machine inputs and outputs.

Reply:
We should also note making sure people can run a local web server?

Reply:
I'm also happy to run a session on Friday or another time this week similar to the sessions that the residents do for the comm labs for the first year ITP students if that would be helpful for getting students up and running on any of the technical assignments (rather than just open office hours).

Reply:
@ellennickles yes, my plan is to spend probably half the class writing the code for basic image classification from scratch reviewing p5 at the same time. I think I will leave teachable machine and transfer learning for Wednesday to not overload the students and have them do a short blog post and exercise before Wednesday.

@joeyklee For now I am going to stick to the web editor entirely. But yes, this will inevitably come up and I'll need to address in class or through a separate help session. I also have this [playlist which covers most of it](https://www.youtube.com/playlist?list=PLRqwX-V7Uu6Zu_uqEA6NqhLzKLACwU74X) and hasn't gone too out of date yet?

@lydiajessup That would be amazing! Feel free to just go ahead and schedule something / announce through the class google group!